### DISCUSSION TITLE
Select multiple paragraphs using match mode, sane `xc` behavior and action without motion first

### DISCUSSION THREAD

**Discussion by linde12** [helix-editor/helix#5616] [Q&A]
Hey. New to helix, coming from many years of vim. I have three questions:

**Q1**
I'm wondering if its possible to select multiple paragraph using match mode (e.g. `map`)

In vim this is possible with e.g. `vap` to enter visual mode and select around paragraph, and then simply doing another `ap` so it becomes `vapap`. Obviously, if you know beforehand how many paragraphs you are going to select, you can also do something like `2vap`

In helix however `mapap` selects the paragraph, then enters insert (from append via `a`) and enters the letter "p"

**Q2**
Is it possible to delete something without selecting first? I know this is kind of against kak and hx principles, but i miss vims e.g. delete in paragraph via `dap` instead of having to `mapd`.

**Q3**
Is it possible to make `xc` behave in a sane way like vims `cc`? I have no idea who thought it'd be a good idea for `xc` to also delete the newline character, but i'd challenge him to a fist fight lol. IMO that effectively makes `xc` superflous, whether it is consistent or not. Right now i'm doing `xdO` to work around this flaw, but it is a real pain point for me which i doubt i will ever get used to. `xc`, apart from deleting the line i wanted to modify, also doesn't put me at the right indentation, making it even less useful.

I like helix. It is very convenient, fast and modern with built-in themes, easy configuration, built-in LSP support, fuzzy file finder etc. which you would expect from a modern editor but i am not sure i like the motion-first concept. It's good when you're new and learning, but when i _know_ what i want to do i feel the the motion-first approach is just a hinderance. I want to say e.g. `ciw` not `miwc` all the time. I know what im doing. I think i can get used to most of it, but my above questions are real pain points for me, hence why i'm asking you guys for help!

I don't mean to offend anyone, i'm a bit frustrated and looking for some advice. Thanks.

**Comment by gabydd** [comment]
for q3 i would suggest customizing the keymap to have a key for ``["extend_to_line_bounds", "trim_selections"]``(``x_``) (this will also keep indentation)

**Comment by sudormrfbin** [comment]
1. You can use `]p` and `[p` after entering extend mode with `v` to select any number of paragraphs.

2. This is explicitly not possible, and there's a vim mode discussion somewhere in the discussion page that describes why a vim mode (i.e. act without a selection first) will never be implemented in helix either (on mobile right now so can't link it).

**Reply by soloturn** [reply]
problem with ][ keys on many international keyboards is that they are behind alt-gr, so painful. the physical key location on english keyboards of course is there - but has a different key. there is no default binding to physical locations and displaying it then as most of the time they go with mnemonics. but then next paragrapf is not `np` or so but includes one of these inaccessible brackets ...

**Reply by pascalkuthe** [reply]
If square brackets are not eaaily accessible on your keyboard that is a problem with your keyboard not helix. 

I would recommend switching to better layout. At least qwert (and if you'd switch anyway you may aswell go all the way and use something like colmak).

Otherwise you can always the keymap but this is hardly an issue with helix.

**Reply by linde12** [reply]
> If square brackets are not eaaily accessible on your keyboard that is a problem with your keyboard not helix.

I think if people have to switch keyboard layouts to use helix comfortably they will simply choose not to use helix. Large parts of the world use non-US keyboard layouts where alt-gr is required to type certain symbols. Not _only_ a helix issue though, and as you said it _can_ be worked around.

Even on english layout, using [] requires you to either use pink/ring finger _or_ move your middle/pointy finger using bad wrist motions. Sticking to more accessible keys just makes more sense to me.

**Comment by linde12** [comment]
Feel free to close this, it is not relevant for me anymore at least. Helix looks promising and there are a lot of things that i love about it (batteries included, little configuration, fast, simple) but the kak philosophy doesnt sit well with me and even then there are too many awkward design decisions with the various motions for me to work around. Can't teach an old dog to sit i guess 😺
